assets/js/rootLantiq.js

```diff
@@ -2,40 +2,80 @@ const LOAD_ADDR = "80800000"
 const IMAGE0_ADDR = "C0000 740000";
 const IMAGE1_ADDR = "800000 800000";
 
-async function detectUboot(reader) {
+const textDecoder = new TextDecoder();
+const textEncoder = new TextEncoder();
+
+async function detectUboot(port) {
+    let reader = undefined;
+
     while (true) {
-        const { value, done } = await reader.read();
+        try {
+            if (reader == undefined) {
+                reader = port.readable.getReader();
+            }
+
+            const { value, done } = await reader.read();
+            const linesValue = textDecoder.decode(value).split('\n');
 
-        if (value.startsWith('U-Boot')) {
-            return;
+            for (const line of linesValue) {
+                if (line.startsWith('U-Boot')) {
+                    return;
+                }
+            }
+        } catch (e) {
+            console.log(e);
+        } finally {
+            if (reader) {
+                reader.releaseLock();
+                reader = undefined;
+            }
         }
     }
 }
 
-async function waitUbootStop(writer, reader, sfpModel, outputMsgCallback) {
+async function waitUbootStop(port, sfpModel, outputMsgCallback) {
+    let writer = port.writable.getWriter();
+    const interruptChar = new Uint8Array([3]);
```

Comment:
despite being a fan of illegible code, I really don't like this line

Reply:
Why? Just create an Array with a single ASCII character with binary value corresponding to 3, it doesn't seem difficult to understand at most I can put a comment on it.

Reply:
Simply if I see String.fromcharcode(3) it can be guessed what it does, this not unless I look at the history. A comment is a good idea